### PR TITLE
AOT upload checkpoints tests

### PR DESCRIPTION
**Pull request by sindhuvahinis**
Adding tests cases for some for SM Training Job.
- Test AOT with and without passing any model_dir argument 
- Invoke AOT by using the command `train`
- Test cases for upload the partitioned checkpoints to S3
- Adding a few testcases for DeepSpeed with Default handler

### CODE REVIEW

File: tests/integration/llm/prepare.py [hunk 1]
```diff
@@ -25,13 +25,32 @@
         "option.s3url": "s3://djl-llm/opt-6b7/",
         "option.tensor_parallel_degree": 4,
         "option.task": "text-generation",
-        "option.dtype": "float16"
+        "option.dtype": "float16",
+        "option.save_mp_checkpoint_path": "/opt/ml/input/data/training/partition-test"
     },
-    "gpt-neox-20b": {
```
Comment (lanking520) [inline tests/integration/llm/prepare.py:30]: why swapping the model?

Comment (sindhuvahinis) [reply tests/integration/llm/prepare.py:30]: Did not swap it. This was added in prepare.py but never used in llm_intergation.yaml.

Comment (sindhuvahinis) [reply tests/integration/llm/prepare.py:30]: Do you want to add gptneox as well in the test cases?

**Review by lanking520 — APPROVED** [review]

File: .github/workflows/llm_integration.yml [hunk 2]
```diff
@@ -456,9 +460,10 @@ jobs:
           sudo rm -rf models
           python3 llm/prepare.py fastertransformer_raw_aot t5-small
           ./launch_container.sh deepjavalibrary/djl-serving:$DJLSERVING_DOCKER_TAG $PWD/models deepspeed \
-          partition --model-dir /opt/ml/model/test/
+          train
```
Comment (siddvenk) [inline .github/workflows/llm_integration.yml:463]: It looks like we support both `train` and `partition` as commands that invoke the partitioning. 

I'm assuming `train` is needed for SageMaker. Once we add SageMaker tests, can we keep the command consistent here?

**Review by siddvenk — APPROVED** [review]

**Pull request merged**